lib/ain-evm/src/core.rs

```diff
@@ -816,6 +816,91 @@ impl EVMCoreService {
 
         Ok((listener.trace, execution_success, return_value, used_gas))
     }
+
+    pub fn create_access_list(&self, arguments: EthCallArgs) -> Result<(AccessList, u64)> {
```

Comment:
trace_transaction and create_access_list should not be placed under block template method implementations. We should not be placing the implementation of these methods anyway inside the core module, and perhaps shift the implementation abstracted inside the executor, while the core service wraps the call to the executor.

Reply:
While I agree, this is beyond the scope for this PR since `create_access_list` uses multiple executors which could introduce some complexity to moving them inside the executor.

Reply:
Resolved in PR #2824